### PR TITLE
Enable nullable on Authentication.MicrosoftAccount

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/aspnetcore/pull/29252

### CODE REVIEW

File: src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs [hunk 1]
```diff
@@ -48,14 +48,14 @@ protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIden
...
-                return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+                return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
             }
         }
 
         /// <inheritdoc />
         protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
         {
-            var queryStrings = new Dictionary<string, string>
+            var queryStrings = new Dictionary<string, string?>
```
Comment (Tratcher) [inline src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs:58]: Why `string?` here?

File: src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs [hunk 2]
```diff
@@ -93,13 +93,13 @@ protected override string BuildChallengeUrl(AuthenticationProperties properties,
         }
 
         private void AddQueryString<T>(
-           IDictionary<string, string> queryStrings,
+           IDictionary<string, string?> queryStrings,
            AuthenticationProperties properties,
            string name,
-           Func<T, string> formatter,
+           Func<T, string?> formatter,
```
Comment (Tratcher) [inline src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs:99]: This formatter shouldn't be returning null if the input was non-null.

**Review by Tratcher — COMMENTED** [review]
I'm confused...

File: src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs (hunk 2 above)
Comment (JamesNK) [reply src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs:99]: But T could be null.

Comment (Tratcher) [reply src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs:99]: We check for that before calling the formatter, though it looks like we would fall back to passing the default to the formatter which might be null. How about adding a check on line 108 to skip the formatter if defaultValue is null?

**Review by Tratcher — APPROVED** [review]

**Pull request merged**